archunit-junit/junit5/engine/src/main/java/com/tngtech/archunit/junit/AbstractArchUnitTestDescriptor.java

```diff
@@ -34,37 +41,74 @@
 import static java.util.stream.Collectors.toSet;
 
 abstract class AbstractArchUnitTestDescriptor extends AbstractTestDescriptor implements Node<ArchUnitEngineExecutionContext> {
-    private final Set<TestTag> tags;
-    private final SkipResult skipResult;
+    private final ConditionEvaluator conditionEvaluator = new ConditionEvaluator();
+    private final AnnotatedElementComposite annotatedElement;
+    private final Map<NamespacedKey, Object> store;
 
-    AbstractArchUnitTestDescriptor(UniqueId uniqueId, String displayName, TestSource source, AnnotatedElement... elements) {
+    AbstractArchUnitTestDescriptor(UniqueId uniqueId, String displayName, TestSource source, AnnotatedElement... annotatedElements) {
         super(uniqueId, displayName, source);
-        tags = Arrays.stream(elements).map(this::findTagsOn).flatMap(Collection::stream).collect(toSet());
-        skipResult = Arrays.stream(elements)
-                .map(e -> AnnotationSupport.findAnnotation(e, ArchIgnore.class))
-                .filter(Optional::isPresent)
-                .map(Optional::get)
+        this.annotatedElement = AnnotatedElementComposite.of(annotatedElements);
+        this.store = new ConcurrentHashMap<>();
+    }
+
+    private boolean shouldBeUnconditionallyIgnored() {
+        return AnnotationUtils.streamAnnotations(annotatedElement, ArchIgnore.class, Disabled.class)
                 .findFirst()
-                .map(ignore -> SkipResult.skip(ignore.reason()))
-                .orElse(SkipResult.doNotSkip());
+                .isPresent();
     }
 
-    private Set<TestTag> findTagsOn(AnnotatedElement annotatedElement) {
-        return AnnotationSupport.findRepeatableAnnotations(annotatedElement, ArchTag.class)
-                .stream()
-                .map(annotation -> TestTag.create(annotation.value()))
+    private Set<TestTag> findTagsOn(AnnotatedElementComposite annotatedElement) {
+        return AnnotationUtils.streamRepeatableAnnotations(annotatedElement, ArchTag.class, Tag.class)
+                .map(annotation -> TestTag.create(ReflectionUtils.invokeMethod(annotation, "value")))
                 .collect(toSet());
     }
 
+    @Override
+    public ArchUnitEngineExecutionContext prepare(ArchUnitEngineExecutionContext context) throws Exception {
+        getExtensionsFromTestSource().forEach(context::registerExtension);
+        return context;
+    }
+
+    @SuppressWarnings("unchecked")
+    protected Collection<Extension> getExtensionsFromTestSource() {
+        return ((Stream<ExtendWith>) AnnotationUtils.streamRepeatableAnnotations(annotatedElement, ExtendWith.class))
```

Comment:
This is a simplified mechanism for detecting JUnit extensions. 

For now, only `EnabledIf` annotations are handled.